src/database/api/DataSnapshot.ts

```diff
@@ -1,187 +1,177 @@
-import { validateArgCount, validateCallback } from "../../utils/validation";
-import { validatePathString } from "../core/util/validation";
-import { Path } from "../core/util/Path";
-import { exportPropGetter } from "../core/util/util";
-import { PRIORITY_INDEX } from "../core/snap/indexes/PriorityIndex";
+import { validateArgCount, validateCallback } from '../../utils/validation';
+import { validatePathString } from '../core/util/validation';
+import { Path } from '../core/util/Path';
+import { PRIORITY_INDEX } from '../core/snap/indexes/PriorityIndex';
+import { Node } from '../core/snap/Node';
+import { Reference } from './Reference';
+import { Index } from '../core/snap/indexes/Index';
+import { ChildrenNode } from '../core/snap/ChildrenNode';
 
 /**
  * Class representing a firebase data snapshot.  It wraps a SnapshotNode and
  * surfaces the public methods (val, forEach, etc.) we want to expose.
- *
- * @constructor
- * @param {!fb.core.snap.Node} node A SnapshotNode to wrap.
- * @param {!Firebase} ref The ref of the location this snapshot came from.
- * @param {!fb.core.snap.Index} index The iteration order for this snapshot
  */
-export const DataSnapshot = function(node, ref, index) {
+export class DataSnapshot {
   /**
-   * @private
-   * @const
-   * @type {!fb.core.snap.Node}
+   * @param {!fb.core.snap.Node} node_ A SnapshotNode to wrap.
```

Comment:
Can we refactor the fb.*.*.* comments? These all have clear analogs in the classes that we expose.